core/src/main/java/com/linecorp/armeria/common/stream/StreamMessage.java

```diff
@@ -143,6 +148,69 @@
         }
     }
 
+    /**
+     * Creates a new {@link StreamMessage} that streams the specified {@link File}.
+     * The default buffer size({@value PathStreamMessage#DEFAULT_FILE_BUFFER_SIZE}) is used to read
+     * the {@link File}.
+     * If the size of a {@link File} is less then {@value PathStreamMessage#DEFAULT_FILE_BUFFER_SIZE},
+     * the returned {@link StreamMessage} will emit only one {@link HttpData}.
+     * If the size of a {@link File} is bigger then {@value PathStreamMessage#DEFAULT_FILE_BUFFER_SIZE},
+     * the returned {@link StreamMessage} will emit {@link HttpData}s chunked to
+     * {@value PathStreamMessage#DEFAULT_FILE_BUFFER_SIZE} except for the last one.
+     *
+     */
+    static StreamMessage<HttpData> of(File file) {
+        requireNonNull(file, "file");
+        return of(file.toPath());
+    }
+
+    /**
+     * Creates a new {@link StreamMessage} that streams the specified {@link Path}.
+     * The default buffer size({@value PathStreamMessage#DEFAULT_FILE_BUFFER_SIZE}) is used to read
+     * the {@link Path}.
+     * If the size of a {@link Path} is less then {@value PathStreamMessage#DEFAULT_FILE_BUFFER_SIZE},
+     * the returned {@link StreamMessage} will emit only one {@link HttpData}.
+     * If the size of a {@link Path} is bigger then {@value PathStreamMessage#DEFAULT_FILE_BUFFER_SIZE},
+     * the returned {@link StreamMessage} will emit {@link HttpData}s chunked to
+     * {@value PathStreamMessage#DEFAULT_FILE_BUFFER_SIZE} except for the last one.
+     */
+    static StreamMessage<HttpData> of(Path path) {
+        requireNonNull(path, "path");
+        return of(path, DEFAULT_FILE_BUFFER_SIZE);
+    }
+
+    /**
+     * Creates a new {@link StreamMessage} that streams the specified {@link Path}.
+     * The specified {@code bufferSize} is used to read the {@link Path}.
+     *
+     * If the size of a {@link Path} is less then {@code bufferSize},
+     * the returned {@link StreamMessage} will emit only one {@link HttpData}.
+     * If the size of a {@link Path} is bigger then {@code bufferSize},
+     * the returned {@link StreamMessage} will emit {@link HttpData}s chunked to
+     * {@code bufferSize} except for the last one.
+     *
+     * @param path the path of the file
+     * @param bufferSize the maximum allowed size of the {@link HttpData} buffers
+     */
+    static StreamMessage<HttpData> of(Path path, int bufferSize) {
+        return of(path, ByteBufAllocator.DEFAULT, bufferSize);
+    }
+
+    /**
+     * Creates a new {@link StreamMessage} that streams the specified {@link Path}.
+     * The specified {@code bufferSize} is used to read the {@link Path}.
+     *
+     * @param path the path of the file
+     * @param alloc the {@link ByteBufAllocator} which will allocate the content buffer
+     * @param bufferSize the maximum allowed size of the {@link HttpData} buffers
+     */
+    static StreamMessage<HttpData> of(Path path, ByteBufAllocator alloc, int bufferSize) {
```

Comment:
How about adding another method that takes `ExecutorService`?

Reply:
That sounds good.